### PR TITLE
Update to iOS 15 deployment target

### PR DESCRIPTION
**Pull request by samsymons**
Task/Issue URL: https://app.asana.com/0/414235014887631/1207384568093628/f
Tech Design URL:
CC:

**Description**:

This PR updates the app to use an iOS 15 deployment target.

I've intentionally avoided cleaning up availability checks to keep this PR small and simple, in case it needs to be reverted last-minute. After the change has been released and looks good, I'll clean up the availability checks.

Each target in the app had overridden the deployment target, so I removed those overrides in favour of having each target inherit the deployment target from the parent project settings. Now we only have to change one value to set the deployment target across the board.

**Workflow Runs**:

* [Alpha Build](https://github.com/duckduckgo/iOS/actions/runs/9751815781)
* [E2E Tests](https://github.com/duckduckgo/iOS/actions/runs/9751819289)

**Steps to test this PR**:
1. Check the the deployment target on each target in the app is iOS 15
2. Check that the app builds and runs fine, and that CI passes

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -185,7 +185,7 @@ jobs:
 
         set -o pipefail && xcodebuild \
         -scheme "DuckDuckGo" \
-        -destination "platform=iOS Simulator,name=iPhone 14" \
+        -destination "platform=iOS Simulator,name=iPhone 15" \
```
Comment (samsymons) [inline .github/workflows/pr.yml:188]: I was getting workflow failures without this change.

**Comment by bwaresiak** [comment]
@samsymons Changes looks ok, maybe we should try running some flows (e2e tests, ad hoc build) from this branch?

**Comment by samsymons** [comment]
@bwaresiak I've updated the description to links of successful workflow runs. The remaining one I haven't tested is the Nightly Alpha workflow, but there's no good way to test it without publishing a real build to the alpha channel, and because the ad-hoc pipeline works (proving the ad-hoc target archives successfully) I feel it's acceptable.

**Review by bwaresiak — APPROVED** [review]

**Comment by bwaresiak** [comment]
@samsymons yes that's great, thank you :)

**Pull request merged**